### PR TITLE
Replace rollaxis() with moveaxis()

### PR DESCRIPTION
**Pull request by Bchass**
### Description

This pull request is to address part of https://github.com/astropy/astropy/issues/15721

Specifically: 

- [x] time
- [x] stats

Fixes #<Issue Number>

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by Bchass** [comment]
Converted to draft to take of:
- modeling (main focus as this one requires the most work)
-  coordinates

File: astropy/time/core.py [hunk 1]
```diff
@@ -2250,8 +2250,8 @@ def light_travel_time(
         )
 
         # Move X,Y,Z to last dimension, to enable possible broadcasting below.
-        cpos = np.rollaxis(cpos, 0, cpos.ndim)
-        spos = np.rollaxis(spos, 0, spos.ndim)
+        cpos = np.moveaxis(cpos, 0, cpos.ndim - 1)
+        spos = np.moveaxis(spos, 0, spos.ndim - 1)
```
Comment (taldcroft) [inline astropy/time/core.py:2254]: I think this should be `np.moveaxis(cpos, 0, -1)`.

**Review by taldcroft — COMMENTED** [review]
Code coverage is showing that this is not tested, so we need to see what's going on there and ensure testing for a scalar and multidimensional case.

**Comment by neutrinoceros** [comment]
Hi @Bchass , thank you for picking this up. I note that #15728 is already aimed at fixing the `coordinates` module so I suggest not to do it here so everyone gets a chance at contributing. Thanks !

File: astropy/time/core.py (hunk 1 above)
Comment (dhomeier) [reply astropy/time/core.py:2254]: Looking at the table from https://numpy.org/doc/stable/reference/generated/numpy.rollaxis.html the two should be equivalent, but I'd agree `-1` would be clearer to read.

Comment (Bchass) [reply astropy/time/core.py:2254]: I'll push changes later tonight for cleaner format.

**Comment by Bchass** [comment]
Did some refactoring for `modeling`. Not sure if `rollaxis` needs to be replaced in `modeling/core.py`. Test cases fail and they can be altered in a way to make them pass so `assert (1, 3, 4) == (1, 4, 3)`

```
pivot = 2
model_set_axis = 1
y = np.moveaxis(y, pivot, model_set_axis)
```



Not sure if that would be the correct approach so I left it alone.

**Comment by dhomeier** [comment]
Thanks; if the tests are reproducing the original axis orders in `modeling` now, `destination` is probably set correctly, but this should be reviewed closely by one of the maintainers.

**Comment by dhomeier** [comment]
With the tests passing what remains to be done is add tests for the 3 changes not covered as indicates by `codecov` – any idea where and how to best test those, @taldcroft ?

**Pull request closed without merging**